server/src/main/java/io/crate/auth/HostBasedAuthentication.java

```diff
@@ -122,6 +125,9 @@ public static HBAConf parse(Settings settings) {
 
     private final Supplier<String> clusterId;
 
+    // JwkProvider are cached to reused in JWTAuthenticationMethod which are stateless and create on a invocation-base
+    private final ConcurrentHashMap<String, JwkProvider> cachedJwkProviders = new ConcurrentHashMap<>();
```

Comment:
The actual caching of the public keys happens within the JwkProvider provided by the library. Since `JWTAuthenticationMethod` have a short lifecycle and only created on per-request-base the actual state is hold here.